src/syncback/mod.rs

```diff
@@ -0,0 +1,495 @@
+mod file_names;
+mod fs_snapshot;
+mod hash;
+mod property_filter;
+mod ref_properties;
+mod snapshot;
+
+use anyhow::Context;
+use memofs::Vfs;
+use rbx_dom_weak::{
+    types::{Ref, Variant},
+    Instance, WeakDom,
+};
+use serde::{Deserialize, Serialize};
+use std::{
+    collections::{HashMap, HashSet, VecDeque},
+    env,
+    path::Path,
+    sync::OnceLock,
+};
+
+use crate::{
+    glob::Glob,
+    snapshot::{InstanceWithMeta, RojoTree},
+    snapshot_middleware::Middleware,
+    syncback::ref_properties::link_referents,
+    Project,
+};
+
+pub use file_names::{extension_for_middleware, name_for_inst, validate_file_name};
+pub use fs_snapshot::FsSnapshot;
+pub use hash::*;
+pub use property_filter::{filter_properties, filter_properties_preallocated};
+pub use ref_properties::collect_referents;
+pub use snapshot::{SyncbackData, SyncbackSnapshot};
+
+/// The name of an enviroment variable to use to override the behavior of
+/// syncback on model files.
+/// By default, syncback will use `Rbxm` for model files.
+/// If this is set to `1`, it will instead use `Rbxmx`. If it is set to `2`,
+/// it will use `JsonModel`.
+///
+/// This will **not** override existing `Rbxm` middleware. It will only impact
+/// new files.
+const DEBUG_MODEL_FORMAT_VAR: &str = "ROJO_SYNCBACK_DEBUG";
+
+/// A glob that can be used to tell if a path contains a `.git` folder.
+static GIT_IGNORE_GLOB: OnceLock<Glob> = OnceLock::new();
+
+pub fn syncback_loop(
```

Comment:
One idea is to prepend the root instance name to ignoreTrees entries when we start the syncback loop.

The ignoreTrees example becomes 
```json
{ "ignoreTrees": [ "ServerStorage/ImportantSecrets" ] }
```
This would make the behavior of ignoreTrees entries more similar to ignorePaths, in that they implicitly start at the root.

Reply:
`ignoreTrees` is honestly something I implemented ages ago and haven't touched all that much recently. I agree this is more intuitive behavior and it's something I wanted to fix.

For context, originally the root of the Roblox file wasn't renamed to match the project's name, so it was generally always `DataModel`. This was... not ideal, but it at least made them work consistently. Now we have no such luck. This is a good catch though, since it slipped my mind!

Reply:
Since this is different from how it works on the Uplift Games fork, do you think it'd be a good idea to document it in the changelog (and also do the same for any other deviations we may make)?

Reply:
It's probably worth noting if/when we make an RC for syncback on `master`, but for the final release I don't think it's necessary. The changelog will need to be updated either way, so I think this is fine.